### DISCUSSION TITLE
How to determine the current branch via gix as library

### DISCUSSION THREAD

**Discussion by sassman** [GitoxideLabs/gitoxide#1303] [Q&A] [answered]
I wonder whats the right way of figuring out whats the branch currently checked out by using gix as a library (from a custom binary)?

**Comment by Ltrlg** [comment, marked as answer]
I think you need a chain of the following calls:
* [`Repository::head`](https://docs.rs/gix/latest/gix/struct.Repository.html#method.head)
* [`Head::referent_name`](https://docs.rs/gix/latest/gix/struct.Head.html#method.referent_name)
* [`FullNameRef::category_and_short_name`](https://docs.rs/gix-ref/0.41.0/gix_ref/struct.FullNameRef.html#method.category_and_short_name) (and filter on category)

**Reply by sassman** [reply]
Thanks, I was able to figure it out.
What also works for me is this:

```rust
let branch = repo
    .head_name()?
    .unwrap_or(FullName::try_from("unknown")?)
    .shorten()
    .to_string();
```
